torch/export/_safeguard.py

```diff
@@ -0,0 +1,15 @@
+import torch
+from torch.overrides import TorchFunctionMode
+
+
+class GradModeUnsupportedSafeguard(TorchFunctionMode):
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        kwargs = kwargs or {}
+        unsupported_grad_mode_ops = [
+            torch._C._set_grad_enabled,
+        ]
+        if func in unsupported_grad_mode_ops and torch._C._get_dispatch_mode(
+            torch._C._TorchDispatchModeKey.PROXY
+        ):
+            raise RuntimeError(f"{func} is not supported for grad mode")
```

Comment:
This error message is bit confusing. Can you change it to something like:
```
RuntimeError(f"Encountered autograd state manager op {func} while exporting. This is unsafe because we don't capture this op in torch.export today hence we can't reflect the user intention soundly")